public/plugin.tsx

```diff
@@ -329,7 +333,24 @@ export class AssistantPlugin
         getDisplayName: () => 'Generate visualization',
         getIconType: () => 'visLine' as const,
         execute: async () => {
-          core.application.navigateToApp(TEXT2VIZ_APP_ID);
+          const dataset = data.query.queryString.getQuery().dataset;
+          const url = new URL(core.application.getUrlForApp(TEXT2VIZ_APP_ID, { absolute: true }));
+          if (dataset && dataset.type === 'INDEX_PATTERN') {
+            url.searchParams.set(INDEX_PATTERN_URL_SEARCH_KEY, dataset.id);
+          }
+          /**
+           * TODO: the current implementation of getting query assistant input needs to be refactored
+           * once query assistant is moved to dashboard-assistant repo. Currently, there is no better
+           * way to get the input value as the query assistant is currently implemented in OSD core.
+           */
+          const queryAssistInputEle = document.getElementsByClassName('queryAssist__input')[0];
+          if (queryAssistInputEle instanceof HTMLInputElement) {
+            const input = queryAssistInputEle.value;
+            if (input) {
+              url.searchParams.set(ASSISTANT_INPUT_URL_SEARCH_KEY, input);
+            }
+          }
```

Comment:
can we get the value from the place triggered this action? If so, we can use context to pass the value.

Reply:
It's a bit tricky, the context menu is updated based on `data.query.queryString.getUpdates$()`, but the query assistant input is not part of it. Didn't find a way to easily update the context when the input changes. I have a todo to refactor this when we moved query assistant to dashboards-assistant plugin.